rmd_files/RNAseq_report.Rmd

```diff
@@ -2124,15 +2130,20 @@ caner_genes_annot.list[["FusionGDB"]] <- read.table(paste(params$ref_data_dir, p
 names(caner_genes_annot.list[["FusionGDB"]]) <- c("Hgene", "HgeneID", "Tgene", "TgeneID", "FGname", "FGID")
 
 
-##### Add refenence cohort name to the sample name
+##### Add reference cohort name to the sample name
 if ( params$dataset_name_incl != "" ) {
   sample_name <- paste0(params$sample_name, "_", params$dataset)
 } else {
   sample_name <- params$sample_name
 }
 
 ##### Read in reference datasets and merge them with sample data. This part outputs a vector with first element containing the merged data and second element containing merged targets info
-ref_dataset.list[[dataset]] <- combineDatasets(sample_name=sample_name, sample_counts=counts, ref_data=ref_dataset, report_dir = results_dir, dataset = dataset)
+if ( exists("counts_salmon") ){
```

Comment:
So the if/else essentially just changes what you feed to the `sample_counts`, and you're duplicating the full function call.
Maybe an alternative could be:

```R
counts1 <- if (exists("counts_salmon") {
  counts_salmon
  } else if (exists("counts")) {
  counts
  } else {
  NULL
  }
  
if (!is.null(counts1)) {
ref_dataset.list[[dataset]] <- combineDatasets(
  sample_name=sample_name, 
  sample_counts=counts1,
  ref_data=ref_dataset, 
  report_dir = results_dir,
  dataset = dataset)
}
  
```

Also note that using `exists` there would look in your R environment for anything under that object name e.g. a loaded pkg with a function `counts` would hit the second if.

Reply:
updated in https://github.com/umccr/RNAsum/commit/70ba9bc292eb02c20b5177e2d51e09b4ffb09826

I have renamed object `counts` to `counts_txi` to make it more specific to this use case. Hopefully, this will avoid the issue.